processor/processor.go

```diff
@@ -485,20 +486,37 @@ func (p *processor) addReference(parent *types.Type, child *types.Type) {
 	}
 }
 
-func mkRegistry() (*markers.Registry, error) {
+func mkRegistry(customMarkers []config.Marker) (*markers.Registry, error) {
 	registry := &markers.Registry{}
-	err := registry.Define(objectRootMarker, markers.DescribesType, true)
-	if err != nil {
+	if err := registry.Define(objectRootMarker, markers.DescribesType, true); err != nil {
 		return nil, err
 	}
 
 	for _, marker := range crdmarkers.AllDefinitions {
-		err = registry.Register(marker.Definition)
-		if err != nil {
+		if err := registry.Register(marker.Definition); err != nil {
 			return nil, err
 		}
 	}
 
+	for _, marker := range customMarkers {
+		t := markers.DescribesField
+		switch marker.Target {
+		case config.TargetTypePackage:
+			t = markers.DescribesPackage
+		case config.TargetTypeType:
+			t = markers.DescribesType
+		case config.TargetTypeField:
+			t = markers.DescribesField
+		default:
+			zap.S().Warnf("Skipping custom marker %s with unknown target type %s", marker.Name, marker.Target)
+			continue
+		}
+
+		if err := registry.Define(marker.Name, t, struct{}{}); err != nil {
```

Comment:
Should we support custom markers with value?

I think that by defining the custom marker with the output type `struct{}{}`, it is not possible to have a custom marker with a value. It is kind of a shame.

Reply:
let's start with no-value marker, we can do that later?

Reply:
Yes, we can do that later.